tests/subsys/pm/device_runtime/src/main.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <ztest.h>
+#include <ksched.h>
+#include <kernel.h>
+#include <pm/pm.h>
+#include <pm/device_runtime.h>
+#include "dummy_driver.h"
+
+#define MAX_TIMES 10
+#define STACKSIZE 1024
+
+/* Semaphore used to synchronize thread A and thread B*/
+K_SEM_DEFINE(sem, 0, 1);
+
+K_THREAD_STACK_DEFINE(threadA_stack, STACKSIZE);
+K_THREAD_STACK_DEFINE(threadB_stack, STACKSIZE);
+
+static const struct device *dev;
+static struct dummy_driver_api *api;
+
+static struct k_thread threadA;
+static struct k_thread threadB;
+
+
+void threadA_func(void *arg1, void *arg2, void *arg3)
+{
+	int ret;
+
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	ret = pm_device_get(dev);
+	zassert_true(ret == 0, "Fail to get device");
+
+	/* Lets allow threadB run */
+	k_sem_give(&sem);
+
+	/* Block waiting for device operation conclude */
+	ret = pm_device_wait(dev, K_FOREVER);
```

Comment:
same here, shouldn't this happen in the driver? The thread need not deal with this, this all needs to be done in the driver.

Reply:
The purpose was testing the api, this is not a sample. Although I get the point, just created wrappers on driver for these functions instead of call them directly.